src/main/java/aspects/InvadersLogger.aj

```diff
@@ -0,0 +1,55 @@
+package aspects;
+
+import util.Logger;
+
+import java.awt.event.KeyEvent;
+
+import actors.Invader;
+import game.InputHandler;
+import game.Invaders;
+import game.InputHandler.Action;
+
+privileged aspect InvadersLogger {
```

Comment:
I should probably explain the keyword `privileged` - it enables an aspect to basically have read-only access to a private field without changing the visibility modifiers. It's kinda hacky but at least it enables encapsulation.